yarn-project/circuits.js/src/structs/tx_context.ts

```diff
@@ -106,6 +106,7 @@ export class TxContext {
      * Contract deployment data.
      */
     public contractDeploymentData: ContractDeploymentData,
+    // TODO(#4256): The following two fields are most likely redundant now and should be removed.
```

Comment:
They are not redundant. We want to ensure that the `tx_context` matches the current chain. 
Say we have a new deployment that is forking the state of the chain. In this case, it would have a new version for new blocks. But if we use the data in the header you cannot match it and will be using the same version/chainid across multiple versions.

Reply:
Closed the issue and removed the comment